### PR TITLE
docs(api): add processAssets hook

### PR DESCRIPTION
**Pull request by chenxsan**
[webpack 5.0.0 beta 17](https://github.com/webpack/webpack/releases/tag/v5.0.0-beta.17) brings some new hooks. This pull request adds info about them.

Also I've added some inline comments so anyone can review it fast.

### CODE REVIEW

File: src/content/api/compilation-hooks.md [hunk 1]
```diff
@@ -464,6 +465,50 @@ The assets have been optimized.
...
+- `PROCESS_ASSETS_STAGE_OPTIMIZE_COUNT` - Optimize the count of existing assets, e.g. by merging them.
+- `PROCESS_ASSETS_STAGE_OPTIMIZE_COMPATIBILITY` - Optimize the compatibility of existing assets, e.g. add polyfills or vendor-prefixes.
+- `PROCESS_ASSETS_STAGE_OPTIMIZE_SIZE` - Optimize the size of existing assets, e.g. by minimizing or omitting whitespace.
+- `PROCESS_ASSETS_STAGE_SUMMARIZE` - Summarize the list of existing assets.
+- `PROCESS_ASSETS_STAGE_DEV_TOOLING` - Add development tooling to assets, e.g. by extracting a SourceMap.
+- `PROCESS_ASSETS_STAGE_OPTIMIZE_TRANSFER` - Optimize the transfer of existing assets, e.g. by preparing a compressed (gzip) file as separate asset.
+- `PROCESS_ASSETS_STAGE_ANALYSE` - Analyse existing assets.
+- `PROCESS_ASSETS_STAGE_REPORT` - Creating assets for reporting purposes.
+
+### `afterProcessAssets`
```
Comment (chenxsan) [inline src/content/api/compilation-hooks.md:506]: It's called in the source code https://github.com/webpack/webpack/blob/255ae13f59ce36e93b6d8c4301e9cd3366140bda/lib/Compilation.js#L1905

File: src/content/api/compilation-hooks.md [hunk 2]
```diff
@@ -464,6 +465,50 @@ The assets have been optimized.
...
+  },
+  (assets) => {
+    delete assets['a.js'];
+  }
+);
+```
+
+There're many stages to use:
+
+- `PROCESS_ASSETS_STAGE_ADDITIONAL` - Add additional assets to the compilation.
```
Comment (chenxsan) [inline src/content/api/compilation-hooks.md:492]: Defined in `types.d.ts` https://github.com/webpack/webpack/blob/a4d06805f2f934132ee14229a73aaa5c532a822b/types.d.ts#L1214

File: src/content/api/compilation-hooks.md [hunk 3]
```diff
@@ -464,6 +465,50 @@ The assets have been optimized.
 - Callback Parameters: `assets`
 
 
+### `processAssets`
+
+`AsyncSeriesHook`
```
Comment (chenxsan) [inline src/content/api/compilation-hooks.md:470]: types defined in https://github.com/webpack/webpack/blob/a4d06805f2f934132ee14229a73aaa5c532a822b/types.d.ts#L979

File: src/content/api/compilation-hooks.md [hunk 4]
```diff
@@ -464,6 +465,50 @@ The assets have been optimized.
...
+  (assets) => {
+    // code here
+  }
+);
+```
+
+There're many stages to use:
+
+- `PROCESS_ASSETS_STAGE_ADDITIONAL` - Add additional assets to the compilation.
+- `PROCESS_ASSETS_STAGE_PRE_PROCESS` - Basic preprocessing of assets.
```
Comment (EugeneHlushko) [inline src/content/api/compilation-hooks.md:493]:
```suggestion
- `PROCESS_ASSETS_STAGE_PRE_PROCESS` - Basic preprocessing of the assets.
```

File: src/content/api/compilation-hooks.md [hunk 5]
```diff
@@ -464,6 +465,50 @@ The assets have been optimized.
...
+- `PROCESS_ASSETS_STAGE_DERIVED` - Derive new assets from existing assets.
```
Comment (EugeneHlushko) [inline src/content/api/compilation-hooks.md:494]:
```suggestion
- `PROCESS_ASSETS_STAGE_DERIVED` - Derive new assets from the existing assets.
```

File: src/content/api/compilation-hooks.md [hunk 6]
```diff
@@ -464,6 +465,50 @@ The assets have been optimized.
...
+- `PROCESS_ASSETS_STAGE_ADDITIONS` - Add additional sections to existing assets, like a banner or initialization code.
```
Comment (EugeneHlushko) [inline src/content/api/compilation-hooks.md:495]:
```suggestion
- `PROCESS_ASSETS_STAGE_ADDITIONS` - Add additional sections to the existing assets e.g. banner or initialization code.
```

File: src/content/api/compilation-hooks.md [hunk 7]
```diff
@@ -464,6 +465,50 @@ The assets have been optimized.
...
+- `PROCESS_ASSETS_STAGE_OPTIMIZE` - Optimize existing assets in a general way.
+- `PROCESS_ASSETS_STAGE_OPTIMIZE_COUNT` - Optimize the count of existing assets, e.g. by merging them.
+- `PROCESS_ASSETS_STAGE_OPTIMIZE_COMPATIBILITY` - Optimize the compatibility of existing assets, e.g. add polyfills or vendor-prefixes.
```
Comment (EugeneHlushko) [inline src/content/api/compilation-hooks.md:498]: I think `vendor prefixes` is more common than `vendor-prefixes`, i'd suggest to go with the common version
```suggestion
- `PROCESS_ASSETS_STAGE_OPTIMIZE_COMPATIBILITY` - Optimize the compatibility of existing assets, e.g. add polyfills or vendor prefixes.
```

File: src/content/api/compilation-hooks.md [hunk 8]
```diff
@@ -464,6 +465,50 @@ The assets have been optimized.
...
+- `PROCESS_ASSETS_STAGE_OPTIMIZE_SIZE` - Optimize the size of existing assets, e.g. by minimizing or omitting whitespace.
+- `PROCESS_ASSETS_STAGE_SUMMARIZE` - Summarize the list of existing assets.
+- `PROCESS_ASSETS_STAGE_DEV_TOOLING` - Add development tooling to assets, e.g. by extracting a SourceMap.
```
Comment (EugeneHlushko) [inline src/content/api/compilation-hooks.md:501]:
```suggestion
- `PROCESS_ASSETS_STAGE_DEV_TOOLING` - Add development tooling to the assets, e.g. by extracting a source map.
```

File: src/content/api/compilation-hooks.md [hunk 9]
```diff
@@ -464,6 +465,50 @@ The assets have been optimized.
...
+- `PROCESS_ASSETS_STAGE_OPTIMIZE_TRANSFER` - Optimize the transfer of existing assets, e.g. by preparing a compressed (gzip) file as separate asset.
+- `PROCESS_ASSETS_STAGE_ANALYSE` - Analyse existing assets.
```
Comment (EugeneHlushko) [inline src/content/api/compilation-hooks.md:503]:
```suggestion
- `PROCESS_ASSETS_STAGE_ANALYSE` - Analyze the existing assets.
```

File: src/content/api/compilation-hooks.md [hunk 10]
```diff
@@ -464,6 +465,50 @@ The assets have been optimized.
...
+- `PROCESS_ASSETS_STAGE_REPORT` - Creating assets for reporting purposes.
```
Comment (EugeneHlushko) [inline src/content/api/compilation-hooks.md:504]:
```suggestion
- `PROCESS_ASSETS_STAGE_REPORT` - Creating assets for the reporting purposes.
```

File: src/content/api/compilation-hooks.md [hunk 11]
```diff
@@ -464,6 +465,50 @@ The assets have been optimized.
...
+
+### `afterProcessAssets`
+
+`SyncHook`
+
+Executed after `processAssets` finished without error.
```
Comment (EugeneHlushko) [inline src/content/api/compilation-hooks.md:510]:
```suggestion
Called after the [`processAssets`](#processassets) hook had finished without error.
```

**Comment by chenxsan** [comment]
@EugeneHlushko Updated.

**Comment by EugeneHlushko** [comment]
Thanks!

**Pull request merged**